java/ql/src/Telemetry/AutomodelEndpointCharacteristics.qll

```diff
@@ -0,0 +1,325 @@
+/**
+ * For internal use only.
+ */
+
+private import java
+private import semmle.code.java.dataflow.DataFlow
+private import semmle.code.java.dataflow.TaintTracking
+private import semmle.code.java.security.PathCreation
+private import semmle.code.java.dataflow.ExternalFlow as ExternalFlow
+private import semmle.code.java.dataflow.internal.FlowSummaryImpl as FlowSummaryImpl
+private import semmle.code.java.security.ExternalAPIs as ExternalAPIs
+private import semmle.code.java.Expr as Expr
+private import semmle.code.java.security.QueryInjection
+private import semmle.code.java.security.RequestForgery
+private import semmle.code.java.dataflow.internal.ModelExclusions as ModelExclusions
+import AutomodelSharedCharacteristics as SharedCharacteristics
+import AutomodelEndpointTypes as AutomodelEndpointTypes
+
+module CandidatesImpl implements SharedCharacteristics::CandidateSig {
+  class Endpoint = DataFlow::ParameterNode;
+
+  class EndpointType = AutomodelEndpointTypes::EndpointType;
+
+  predicate isNegative(AutomodelEndpointTypes::EndpointType t) {
+    t instanceof AutomodelEndpointTypes::NegativeSinkType
+  }
+
+  // Sanitizers are currently not modeled in MaD. TODO: check if this has large negative impact.
```

Comment:
I think this is acceptable for this PR, no need to resolve this before merging.

Reply:
agree! thx :)